src/services/infra/InfraService.ts

```diff
@@ -229,6 +260,159 @@ export default class InfraService {
+       * has started the site launch flow. This prevents users
+       * from retrying and abusing the system.
+       *
+       * The ability to retry after a failed launch is
+       * tracked in IS-273
+       */
+      return this.launchesService.updateDbForLaunchStart(siteName).andThen(() =>
+        ResultAsync.fromPromise(
+          this.launchSite(
+            user,
+            agency,
+            siteName,
+            primaryDomain,
+            subDomainSettings
+          ),
+          (error) => new SiteLaunchError(`Error launching site: ${error}`)
+        ).andThen((siteLaunch) => {
+          if (siteLaunch.isOk()) {
+            return okAsync(siteLaunch.value)
+          }
+          return errAsync(siteLaunch.error)
+        })
+      )
+    })
+  }
+
+  getGeneratedDnsRecords = (
+    siteName: string
+  ): ResultAsync<
+    DnsResultsForSite,
+    MissingSiteError | AmplifyError | SiteLaunchError
+  > =>
+    this.sitesService
+      .getBySiteName(siteName)
+      .andThen(() => this.launchesService.getDNSRecords(siteName))
+
+  getSiteLaunchStatus(
+    sessionData: UserWithSiteSessionData
+  ): ResultAsync<
+    SiteLaunchDto,
+    MissingSiteError | AmplifyError | SiteLaunchError
+  > {
+    const { siteName } = sessionData
+    // 1. Check if ddb entry for the site exists
+    return (
+      this.dynamoDBService
+        .getLaunchStatus(siteName)
+        // NOTE: as long as the entry exists, we treat it as launching
+        .andThen(() =>
+          this.getGeneratedDnsRecords(siteName).andThen((records) => {
+            if (records.dnsRecords.length === 0) {
+              // will not occur, as the function should not return
+              // an empty array but we defensively check for this
+              return errAsync(
+                new SiteLaunchError(`No DNS records found for ${siteName}`)
+              )
+            }
+            return okAsync({
+              siteStatus: "LAUNCHING",
+              dnsRecords: records.dnsRecords,
+              siteUrl: records.siteUrl,
+            } as SiteLaunchDto)
+          })
+        )
+
+        // 2. If not, get the source of truth from the db
+        .orElse(() =>
+          this.sitesService.getBySiteName(siteName).andThen((site) => {
+            if (!site) {
+              errAsync(
+                new MissingSiteError(
+                  `Site launch status for ${siteName} is unknown`
+                )
+              )
+            }
+            if (site.siteStatus !== SiteStatus.Launched) {
+              return okAsync({
+                siteStatus: "NOT_LAUNCHED",
+              } as SiteLaunchDto)
+            }
+
+            // 3. If site is either launched or in the
+            //    process of being launched, get the dns records
+            return (
+              this.getGeneratedDnsRecords(siteName)
+                .andThen((records) => {
+                  if (records.dnsRecords.length === 0) {
```

Comment:
from the comment here, it sounds like we need to check the `SiteStatus` for it being in the process of being launched also? the earlier check only covers `LAUNCHED`